### ISSUE TITLE
Ability to omit form fields.

### ISSUE THREAD

**Issue by howardroark** [macek/jquery-serialize-object#51]
Hey!  Great tool.  I can't believe jquery does not have this feature.  Must be stuck in Web 2.0 ;)

I think this issue is like #46 but I'm not sure.  Essentially I am using  [`stripe.js`](https://stripe.com/docs/stripe.js) to convert credit card data to a hash for my application to use. I would like to pass my API all the form details except for the fields related to credit card data.  

It would be awesome if this tool could omit form fields that were marked with a particular class or data attribute.

If you point me in the right direction I could take a stab at it as well.

Thanks!

**Comment by malles** [comment]
I think that can be done in one of the magical regexes...
I don't dare touch them, though :)

**Comment by howardroark** [comment]
I think you could just do this....

```
  FormSerializer.serializeObject = function serializeObject() {
    if (this.length > 1) {
      return new Error("jquery-serialize-object can only serialize one form at a time");
    }
    var filteredForm = this.find(':input').not('.omit');
    return new FormSerializer($, this).
      addPairs(filteredForm.serializeArray()).
      serialize();
  };
```

vs this...

```
  FormSerializer.serializeObject = function serializeObject() {
    if (this.length > 1) {
      return new Error("jquery-serialize-object can only serialize one form at a time");
    }
    return new FormSerializer($, this).
      addPairs(this.serializeArray()).
      serialize();
  };
```

**Comment by macek** [comment]
> I think that can be done in one of the magical regexes...
I don't dare touch them, though :)

@malles the regexps are not meant to filter specific fields. Modifications to the regexps are only there to differentiate how various inputs are handled while serializing.

@howardroark I suspect this can be accomplished with minimal modification to the plugin itself

```js
$("#my-form :input").not(".omit").serializeObject();
```
I like this approach because of the inversion of control. From the end user's point of view, there's no additional API/convention to learn and it's compatible with any kind of user-created form. (see also #23).
 
To support this, we'd have to remove the following check in the code you mentioned above

```js
if (this.length > 1) {
  return new Error("jquery-serialize-object can only serialize one form at a time");
}
```

After that, we're just calling `this.serializeArray()` which works on any set of inputs.

Add a couple tests to verify everything works as intended and we'd be good to go.

Thoughts ?

**Comment by howardroark** [comment]
I like the idea of having it as open as possible as well! Have a look at #53 :)

**Comment by mnpenner** [comment]
`$.serializeArray()` does *not* appear to work unless you give a `form` object. I too would like to be able to serialize a sub-section of my form.

**Comment by macek** [comment]
@mnpenner this is something I think that should be supported as well. I'll get to some of these issues as soon as I have a little extra time. Things have just been really hectic for me lately.

**Comment by csch0** [comment]
yes please @macek

**Comment by macek** [comment]
I just published version `2.5.0` which allows you to serialize multiple elements

Please read the specified behavior in [selector-test.js](https://github.com/macek/jquery-serialize-object/blob/master/test/integration/selector-test.js) for more info.
